paper/src/main/java/net/william278/huskhomes/util/FoliaTaskRunner.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of HuskHomes, licensed under the Apache License 2.0.
+ *
+ *  Copyright (c) William278 <will27528@gmail.com>
+ *  Copyright (c) contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package net.william278.huskhomes.util;
+
+import io.papermc.paper.threadedregions.scheduler.GlobalRegionScheduler;
+import io.papermc.paper.threadedregions.scheduler.RegionScheduler;
+import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
+import net.william278.huskhomes.BukkitHuskHomes;
+import net.william278.huskhomes.position.Location;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import static net.william278.huskhomes.PaperHuskHomes.folia;
+import static org.bukkit.Bukkit.getServer;
+
+public interface FoliaTaskRunner extends TaskRunner {
+
+    default RegionScheduler getRegionScheduler() {
+        if (folia) {
+            return getServer().getRegionScheduler();
+        } else {
+            return null;
+        }
+    }
+
+    default GlobalRegionScheduler getGlobalRegionScheduler() {
+        if (folia) {
+            return getServer().getGlobalRegionScheduler();
+        } else {
+            return null;
+        }
+    }
+
+    ArrayList<ScheduledTask> scheduledTasks = new ArrayList<>();
+
+    default int runAsync(@NotNull Runnable runnable, Location location) {
+        // For folia
+        if (getGlobalRegionScheduler() != null) {
+            if (location != null) {
```

Comment:
I did this so that it would be possible to set the location, since 1 method uses it, otherwise there is a chance to get UnsuportedOperationExeption or similar

Reply:
In other cases, null is passed, where it is not necessary to use RegionScheduler

Reply:
> I did this so that it would be possible to set the location, since 1 method uses it, otherwise there is a chance to get UnsuportedOperationExeption or similar
> In other cases, null is passed, where it is not necessary to use RegionScheduler

Which method? It looks like you're always passing `null` as the Location -- I can't see any instance where you're passing a concrete location to a `runAsync` or `runSync` :)